File: Bottom Sheet/app/src/main/java/junseong/k/myapplication/MainActivity.java
package junseong.k.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.core.content.res.ResourcesCompat;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.android.material.bottomsheet.BottomSheetBehavior;
import com.google.android.material.bottomsheet.BottomSheetDialog;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{
    Button btn;
    CoordinatorLayout coordinatorLayout;
    BottomSheetBehavior<View> persistentBottomSheet;
    BottomSheetDialog modalBottomSheet;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        coordinatorLayout=findViewById(R.id.coordinator);
        btn=findViewById(R.id.button);
        btn.setOnClickListener(this);

        initPeristentBottomSheet();
    }

    private void initPeristentBottomSheet(){
        View bottomSheet=coordinatorLayout.findViewById(R.id.bottm_sheet);
        persistentBottomSheet= BottomSheetBehavior.from(bottomSheet);
    }

    @Override
    public void onClick(View v) {
        createDialog();
    }
    private void createDialog(){
        List<DataVO> list=new ArrayList<>();

        DataVO vo=new DataVO();
        vo.title="Keep";
        vo.image= ResourcesCompat.getDrawable(getResources(), R.drawable.ic_lab4_1, null);
        list.add(vo);

        vo=new DataVO();
        vo.title="Inbox";
        vo.image= ResourcesCompat.getDrawable(getResources(), R.drawable.ic_lab4_2, null);
        list.add(vo);

        vo=new DataVO();
        vo.title="Messanger";
        vo.image= ResourcesCompat.getDrawable(getResources(), R.drawable.ic_lab4_3, null);
        list.add(vo);

        vo=new DataVO();
        vo.title="Google+";
        vo.image= ResourcesCompat.getDrawable(getResources(), R.drawable.ic_lab4_4, null);
        list.add(vo);

        RecyclerViewAdpater adapter=new RecyclerViewAdpater(list);
        View view=getLayoutInflater().inflate(R.layout.modal_sheet, null);
        RecyclerView recyclerView=view.findViewById(R.id.recyclerview);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(adapter);

        modalBottomSheet=new BottomSheetDialog(this);
        modalBottomSheet.setContentView(view);
        modalBottomSheet.show();
    }
}

File: SingleChoiceListView/app/src/main/java/junseong/kotlin/singlechoicelistview/MainActivity.kt
package junseong.kotlin.singlechoicelistview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var data1= Array(10){""}
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        for(i in 0 until 10)
            data1[i]="항목${i}"

        val adapter1=ArrayAdapter(this,android.R.layout.simple_list_item_single_choice,data1)
        list.adapter=adapter1
        list.choiceMode=ListView.CHOICE_MODE_SINGLE
        list.setItemChecked(2,true)
        button.setOnClickListener {
            text1.text="${data1[list.checkedItemPosition]} 항목 선택되어있슴다"
        }
    }
}
